salt/modules/dracr.py

```diff
@@ -150,6 +150,43 @@ def __execute_ret(command, host=None,
 
     return cmd
 
+def get_property(host=None,
+                     admin_username=None, admin_password=None, property=None):
+    '''
+    Return specific property
+    '''
+    if property is None:
+        raise SaltException('No property specified!')
+    ret = __execute_ret('get %s' % property, host=host,
+                        admin_username=admin_username,
+                        admin_password=admin_password)
+    return ret
+
+def set_property(host=None,
+                     admin_username=None, admin_password=None, property=None, value=None):
+    '''
+    Set specific property
+    '''
+    if property is None:
+        raise SaltException('No property specified!')
+    elif value is None:
+        raise SaltException('No value specified!')
+    ret = __execute_ret('set %s %s' % (property, value), host=host,
+                        admin_username=admin_username,
+                        admin_password=admin_password)
+    return ret
+
+def property_present(host=None, admin_username=None, admin_password=None, property=None, value=None):
```

Comment:
isn't this method confusing? where would you use it?

if the 'set' is expensive and you want to avoid setting it if it's the same value, the set_property could do a get and do a proper set only if it makes sense.

Reply:
The idea was to distinguish set and get method and if you want to use it as module
```
salt 'server' idrac.property_present' key value
```
you can which could be practical. In state only get and set methods are used because we need to know if value is set or should be set because of test. Important is that get and set methods are equally expensive so this should not matter.

If the name is confusing I can rename it but I did not come up with better name (suggestions are welcome).

Reply:
I think it's confusing because it's inside a module. It would make more sense in a state?

Typically modules are 'actions', and that method doesn't really qualify with an action, but more with a state.

If you have an analogy with the drac.py module and state, you can see that the 'present' python method is inside the state.

Reply:
I find it confusing because in case I want to check if a property is set, I do a get_property. If I want to do a set, I do a set_property.  I assume tests can and do that too.

The only 'present'-like methods I saw were in salt/state/*, which basically you do with the property().

In case you really want it, maybe it makes sense to rename it to something like: set_property_if_doesnt_exist_or_has_different_value, but how is that different than just using the normal set_property, except is more expensive? (potentially doubling the runtime)

If it's just me with this opinion, feel free to ignore it. I'm not requiring changes or approving. I'm still fairly new with salt.

Reply:
I also find the naming of this confusing but the implementation itself seems useful and seems like it makes sense in an execution module.